src/cache/cache_las.c

```diff
@@ -1270,9 +1281,18 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
             WT_ERR(cursor->next(cursor));
         }
 
-        /* Don't remove the non obsolete entry. */
-        if (las_timestamp > S2C(session)->txn_global.oldest_timestamp ||
-          las_timestamp > S2C(session)->txn_global.last_ckpt_timestamp ||
+        /*
+         * There are several conditions that need to be met
+         * before we choose to remove a lookaside entry:
+         *  * If there exists a last checkpoint timestamp, then the lookaside record timestamp
+         *    must be less than last checkpoint timestamp.
+         *  * The entry is globally visible.
+         *  * The entry wasn't from a prepared transaction.
+         */
+        if (((S2C(session)->txn_global.last_ckpt_timestamp != WT_TS_NONE) ?
```

Comment:
I could be wrong here, is this if condition simplified to:
```
if (((S2C(session)->txn_global.last_ckpt_timestamp != WT_TS_NONE) &&
      (las_timestamp > S2C(session)->txn_global.last_ckpt_timestamp)) ||
     !__wt_txn_visible_all(session, las_txnid, durable_timestamp) ||
     prepare_state == WT_PREPARE_INPROGRESS)
```

Reply:
Yes, Fixed.